From 3fb7b66aa38e2d76a52266aaf123db3984765b17 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Pierret=20=28fepitre=29?=
 <frederic.pierret@qubes-os.org>
Date: Fri, 14 Aug 2020 11:39:36 +0200
Subject: [PATCH 2/2] uapi/linux/power_supply.h: add
 power_supply_is_str_property func
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Frédéric Pierret (fepitre) <frederic.pierret@qubes-os.org>
---
 include/uapi/linux/power_supply.h | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/include/uapi/linux/power_supply.h b/include/uapi/linux/power_supply.h
index 952f1b1433da..88c7db04904e 100644
--- a/include/uapi/linux/power_supply.h
+++ b/include/uapi/linux/power_supply.h
@@ -237,4 +237,18 @@ static inline _Bool power_supply_is_watt_property(enum power_supply_property psp
 	return 0;
 }
 
+static inline _Bool power_supply_is_str_property(enum power_supply_property psp)
+{
+	switch (psp) {
+	case POWER_SUPPLY_PROP_MODEL_NAME:
+	case POWER_SUPPLY_PROP_MANUFACTURER:
+	case POWER_SUPPLY_PROP_SERIAL_NUMBER:
+		return 1;
+	default:
+		break;
+	}
+
+	return 0;
+}
+
 #endif /* __UAPI_LINUX_POWER_SUPPLY_H__ */
-- 
2.25.4

